src/librustc/session/mod.rs

```diff
@@ -948,12 +948,14 @@ impl Session {
         self.opts.debugging_opts.teach && self.parse_sess.span_diagnostic.must_teach(code)
     }
 
-    /// Are we allowed to use features from the Rust 2018 edition?
-    pub fn rust_2018(&self) -> bool {
-        self.opts.edition >= Edition::Edition2018
-    }
-
-    pub fn edition(&self) -> Edition {
+    /// What is the edition of the "local crate" being compiled?
+    ///
+    /// You should not call this except as a last resort. It is better
+    /// to do `span.edition()` instead, which gives the edition for a
+    /// particular span: that way, when you are looking at code
+    /// creating a macro from a Rust 2015 crate, you will use the Rust
+    /// 2015 Edition rules.
+    pub fn local_edition(&self) -> Edition {
```

Comment:
We can probably just implement `Default` for `Edition`.
"Local edition" needs to be [kept in global data](https://github.com/rust-lang/rust/pull/50307/files#diff-63d14a8aaf73f01854db110ad690e15dR155) anyway because it's an "edition of `SyntaxContext(0)` aka not-a-macro-expansion" and it's set from default-default `Edition2015` to `self.opts.edition` [immediately after options are parsed](https://github.com/rust-lang/rust/pull/50307/files#diff-707a0eda6b2f1a0537abc3d23133748cR469).

Reply:
Hmm. True, though I'm not wild about the fact that we rely so heavily on global / TLS data for this stuff.